src/_express/handler.js

```diff
@@ -1,62 +1,55 @@
-
-  // make event object look like APIG 1.0
-  // until aws-serverless-express supports APIG 2.0
-  event.path = event.requestContext.http.path;
-  event.method = event.requestContext.http.method;
-  event.httpMethod = event.requestContext.http.method;
-  // APIG 2.0 extracts cookies from headers automatically.
-  // We need to put `cookie` back to `headers.cookie` so it works with current aws-serverless-express
-  if (event.cookies && event.cookies.length > 0) {
-    event.headers.cookie = event.cookies.join('; ');
-  }
+// remove AWS Lambda default handling of unhandled rejections
+// this makes sure dev mode cli catches unhandled rejections
+process.removeAllListeners('unhandledRejection');
 
-  // NOTICE: require() is relative to this file, while existsSync() is relative to the cwd, which is the root of lambda
-  let app;
-  if (fs.existsSync('./app.js')) {
-    // load the user provided app
-
-    try {
-      // eslint-disable-next-line import/no-unresolved
-      app = require('../app.js');
-    } catch (e) {
-      if (e.message.includes("Cannot find module 'express'")) {
-        // user probably did not run "npm i". return a helpful message.
-        return {
-          statusCode: 404,
-          body:
-            'The "express" dependency was not found. Did you install "express" as a dependency within your source folder via npm?',
-        };
-      }
-      // some other require error
-      return {
-        statusCode: 404,
-        body: e.message,
-      };
-    }
-  } else {
-    // load the built-in default app
-    app = require('../_src/app.js');
-  }
+let app = (_req, res) => {
+  res.statusCode = 200;
+  res.end(`Request received: ${_req.method} - ${_req.path}`);
+};
+
+try {
+  // eslint-disable-next-line import/no-unresolved
+  app = require('..');
 
-  if (typeof app !== 'function') {
-    // make sure user exported app in app.js or return a helpful message.
-    return {
-      statusCode: 404,
-      body:
-        'Express app not found. Please make sure you are exporting express from an "app.js" file.',
+  // check if the user forgot to export their express app
+  if (typeof app === 'object' && Object.keys(app).length === 0) {
+    app = (_req, res) => {
+      const e = new Error(
+        'App not found. Please make sure the app is probably exported from the JS file.'
+      );
+      console.error(e);
+      res.statusCode = 404;
+      res.end(e.stack);
     };
   }
+} catch (e) {
+  // return require error and log it for dev mode
+  app = (_req, res) => {
+    console.error(e);
+    res.statusCode = 500;
+    res.end(e.stack);
+  };
+}
+
+// default error handler logs errorss for dev mode
+if (typeof app.use === 'function') {
+  // eslint-disable-next-line
+  app.use((err, req, res, next) => {
+    console.error(err.stack);
+    throw err;
+  });
+}
+
+const handle = serverlessHttp(app);
+
+exports.handler = async (event, context) => {
+  context.callbackWaitsForEmptyEventLoop = false;
```

Comment:
This isn't needed because you're not using the callback

Reply:
oh right, this was taken from aws-serverless-express